refactor(Feature): FeatureCollection extends by Object3D and use local Matrix

gchoqueux opened the pull request:
## Description
`FeatureCollection` uses local transform matrix. All data needs to be transformed to be  in `crs` projection.

The data (`extent` or `Coordinates`) are stored in a local system.
To use `Feature` vertices or `FeatureCollection/Feature` extent in crs projection,
it's necessary to apply `FeatureCollection.matrixWorld` on feature `Coordinates` or `Extent`.

## BREACKING CHANGES:
FeatureCollection/Feature/FeatureGeometry use local transform matrix.

## Motivation and Context
This PR makes the `FeatureCollection` transformations consistent. They are compatible with THREE js transformations.

In `Feature2Texture` the transformation of feature vertices to coordinates texture uses these transformation matrices.  It will be easier to switch from 2d canvas to 3d canvas.

gchoqueux committed:
fix(test): use Extent.applyMatrix4 and Coordinates.applyMatrix4.

mgermerie commented on src/Converter/Feature2Texture.js:
```diff
@@ -166,25 +171,38 @@ export default {
+                canvasSize.set(ctx.canvas.width, ctx.canvas.height, 1.0);
+            }
+
+            world2texture.compose(worldTextureOrigin, quaternion, dimension.divide(canvasSize)).invert();
```
Should the `invert()` be removed, you could try :
```js
const scale = canvasSize.divide(dimension);
world2texture.compose(
    worldTextureOrigin.multiply(scale).negate(),
    quaternion,
    scale,
);
```
Assumed the quaternion is allways neutral (i.e. no rotation), it gives the same result than : 
```js
world2texture.compose(worldTextureOrigin, quaternion, dimension.divide(canvasSize)).invert()
```

gchoqueux replied:
yes, i changed

gchoqueux committed:
refactor(Feature): FeatureCollection extends by Object3D and use local transform matrix.

FeatureCollection uses local transform matrix. All data needs to be transformed to be crs projection.

The data (`extent` or `Coordinates`) are stored in a local system.
To use `Feature` vertices or `FeatureCollection/Feature` extent in crs projection,
it's necessary to apply `FeatureCollection.matrixWorld` on feature `Coordinates` or `Extent`.

BREACKING CHANGES:
FeatureCollection/Feature/FeatureGeometry use local transform matrix.
